### ISSUE TITLE
SHKFacebook does not recognize link or title

### ISSUE THREAD

**Issue by mattdanger** [ideashower/ShareKit#344]
While testing the latest version of my app I found that ShareKit was no longer sharing Facebook links correctly. I believe I have SHKConfig.h configured properly. The user is able to login/authenticate with Facebook correctly but the sharing screen appears with a blank text box and no link or title that I defined in the SHKItem.

Screen shot: http://i.imgur.com/5shJ1l.jpg

Here's my code: 

NSURL *url = [NSURL URLWithString:@"http://example.com"];
NSString *title = [NSString stringWithFormat:@"Check out this link."];
SHKItem *item = [SHKItem URL:url title:title];

[SHKFacebook shareItem:item];

To eliminate the above code as a problem I tried using ShareKit's example actionsheet code found in the install docs. Same problem. 

Nothing in the logs with ShareKit debugging turned on.

Any suggestions for further troubleshooting?

**Comment by iBasel** [comment]
Did you solve this? I'm facing the same problem, but I noticed that if you have the user info saved (so the login page doesn't show), it works correctly.

**Comment by aneesham** [comment]
It doesnt show up in the text box, but the links present after posting, check it.

**Comment by iBasel** [comment]
No, even on my Facebook page, it only shows the text I type in the text box, without a link... and if I post it without text in the text box, it posts something on my wall without any text or link.

**Comment by aneesham** [comment]
Are you using a proxy to connect to the internet? That could be an issue.
Are you testing on device or the Simulator?

**Comment by aneesham** [comment]
SHKItem _item = [SHKItem URL:externalURL title:[NSString stringWithFormat:@" %@",self.title]];
you should see your *title_ as a hyperlink.
Try it.

**Comment by iBasel** [comment]
Thank you for your replies. I tried on both simulator and device, without a proxy, it didn't work. I also tried your code, nothing. I will try to debug the source code tonight and see why it only works when user info is saved.

**Comment by iBasel** [comment]
I found out what causes the problem, but I can't solve it yet. I'm setting a timer to check if the user has logged in to facebook using "isServiceAuthorized" to enable a webview to like our page on fb, without calling this method, it works correctly.

**Comment by mattdanger** [comment]
I have not yet solved this issue. The problem was troubleshooted both in the simulator and on the device. The item did not post with a link as @aneesham suggested.

I have since dropped this version of ShareKit for the more current and supported version: https://github.com/ShareKit/ShareKit
